### PR TITLE
Upgrade Python pre-commit tools

### PR DESCRIPTION
**Pull request by illia-v**
A plugin in my editor has been reformatting unchanged lines annoyingly, it turns out it used a more recent version of black, so I decided to update all the pre-commit tools and commit the produced changes

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,4 +1,4 @@
 [flake8]
-ignore = E501, E203, W503, W504
+ignore = E501, E203, W503, W504, E704
```
Comment (illia-v) [inline setup.cfg:2]: flake8's E704 didn't like this kind of changes by black

Comment (pquentin) [reply setup.cfg:2]: Yes, it should be disabled too: https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html#e701-e704

**Review by pquentin — APPROVED** [review]
Thanks! LGTM

**Comment by sigmavirus24** [comment]
If I may be so bold, pre-commit.ci will auto-run and fix things on PRs from contributors and auto-upgrade dependencies on Mondays. It's also provided free of charge for F/OSS projects

**Pull request merged**